app/controllers/income/estimatedPay/update/IncomeUpdatePayslipAmountController.scala

```diff
@@ -50,46 +50,56 @@ class IncomeUpdatePayslipAmountController @Inject()(
+          mandatorySeqEither match {
+            case Right(mandatorySeq) =>
+              val viewModel = {
+                val employer = IncomeSource(mandatorySeq(0).toInt, mandatorySeq(1))
 
-            val payPeriod = optionalSeq(0)
-            val payPeriodInDays = optionalSeq(1)
-            val totalSalary = optionalSeq(2)
+                val payPeriod = optionalSeq(0)
+                val payPeriodInDays = optionalSeq(1)
+                val totalSalary = optionalSeq(2)
 
-            val errorMessage = "tai.payslip.error.form.totalPay.input.mandatory"
+                val errorMessage = "tai.payslip.error.form.totalPay.input.mandatory"
 
-            val paySlipForm = PayslipForm.createForm(errorMessage).fill(PayslipForm(totalSalary))
+                val paySlipForm = PayslipForm.createForm(errorMessage).fill(PayslipForm(totalSalary))
 
-            PaySlipAmountViewModel(paySlipForm, payPeriod, payPeriodInDays, employer)
-          }
+                PaySlipAmountViewModel(paySlipForm, payPeriod, payPeriodInDays, employer)
+              }
+
+              Ok(views.html.incomes.payslipAmount(viewModel))
 
-          Ok(views.html.incomes.payslipAmount(viewModel))
+            case Left(_) =>
+              Redirect(controllers.routes.TaxAccountSummaryController.onPageLoad())
+          }
         }
       }
   }
 
   def handlePayslipAmount: Action[AnyContent] = (authenticate andThen validatePerson).async { implicit request =>
     implicit val user = request.taiUser
 
-    val employerFuture = IncomeSource.create(journeyCacheService)
+    val incomeSourceFuture = IncomeSource.create(journeyCacheService)
 
     val result: Future[Future[Result]] = for {
-      employer        <- employerFuture
-      payPeriod       <- journeyCacheService.currentValue(UpdateIncome_PayPeriodKey)
-      payPeriodInDays <- journeyCacheService.currentValue(UpdateIncome_OtherInDaysKey)
+      incomeSourceEither <- incomeSourceFuture
+      payPeriod          <- journeyCacheService.currentValue(UpdateIncome_PayPeriodKey)
+      payPeriodInDays    <- journeyCacheService.currentValue(UpdateIncome_OtherInDaysKey)
     } yield {
       val errorMessage = GrossPayPeriodTitle.title(payPeriod, payPeriodInDays)
       PayslipForm
         .createForm(errorMessage)
         .bindFromRequest()
         .fold(
           formWithErrors => {
-            val viewModel = PaySlipAmountViewModel(formWithErrors, payPeriod, payPeriodInDays, employer)
-            Future.successful(BadRequest(views.html.incomes.payslipAmount(viewModel)))
+            incomeSourceEither match {
+              case Right(incomeSource) =>
+                val viewModel = PaySlipAmountViewModel(formWithErrors, payPeriod, payPeriodInDays, incomeSource)
+                Future.successful(BadRequest(views.html.incomes.payslipAmount(viewModel)))
+              case Left(_) => Future.successful(Redirect(controllers.routes.TaxAccountSummaryController.onPageLoad()))
```

Comment:
This will redirect you to the start of the journey if you enter incorrect values and mongo is down, not sure if we would prefer some content here instead, best to have a chat with a BA and UX

Reply:
Asked Andy, waiting for a response